test/lib/factories/WorkspaceFactory.test.ts

```diff
@@ -0,0 +1,184 @@
+			},
+			{scenario: 'a single folder', expectation: 'just that folder', input: ['undotted-directory-a']},
+			{
+				scenario: 'a list of folders',
+				expectation: 'just those folders',
+				input: ['undotted-directory-a', '.dotted-directory-b']
+			},
+			{
+				scenario: 'mixed folders and files',
+				expectation: 'just those files/folders',
+				input: ['.dotted-file-1.txt', path.join('.', 'undotted-directory-b')]
+			}
+		])('Given $scenario, requests a workspace with $expectation', async ({input}) => {
+			// ==== TEST SETUP ====
+			const config: CodeAnalyzerConfig = CodeAnalyzerConfig.withDefaults();
+			const core: CodeAnalyzer = new CodeAnalyzer(config);
+			const factory: WorkspaceFactory = new WorkspaceFactory();
+
+			// ==== TESTED BEHAVIOR ====
+			const workspace: Workspace = await factory.create(core, input);
+
+			// ==== ASSERTIONS ====
+			const workspaceItems: string[] = workspace.getFilesAndFolders();
+			const expectedWorkspaceSet = new Set(input.map(i => path.resolve('.', i)));
+			expect(workspaceItems).toHaveLength(expectedWorkspaceSet.size);
+			for (const expectedWorkspaceItem of expectedWorkspaceSet.keys()) {
+				expect(workspaceItems).toContain(expectedWorkspaceItem);
+			}
+		});
+	});
+
+	describe('Glob matching', () => {
+		it.each([
+			{
+				scenario: 'a single relative file',
+				expectation: 'just that file',
+				input: path.join('.', 'undotted*1.txt'),
+				output: new Set(['undotted-file-1.txt'])
+			},
+			{
+				scenario: 'a list of relative files',
+				expectation: 'just those files',
+				input: path.join('.', 'undotted*'),
+				output: new Set(['undotted-file-1.txt', 'undotted-file-2.txt'])
+			},
+			{
+				scenario: 'a single absolute file',
+				expectation: 'just that file',
+				input: path.resolve(PATH_TO_WORKSPACE, 'undotted*1.txt'),
+				output: new Set(['undotted-file-1.txt'])
+			},
+			{
+				scenario: 'a list of absolute files',
+				expectation: 'just those files',
+				input: path.resolve(PATH_TO_WORKSPACE, 'undotted*'),
+				output: new Set(['undotted-file-1.txt', 'undotted-file-2.txt'])
+			}
+		])('Given a glob that matches $scenario, requests a workspace with $expectation', async ({input, output}) => {
+			// ==== TEST SETUP ====
+			const config: CodeAnalyzerConfig = CodeAnalyzerConfig.withDefaults();
+			const core: CodeAnalyzer = new CodeAnalyzer(config);
+			const factory: WorkspaceFactory = new WorkspaceFactory();
+
+			// ==== TESTED BEHAVIOR ====
+			const workspace: Workspace = await factory.create(core, [input]);
+
+			// ==== ASSERTIONS ====
+			const workspaceBaseNames: string[] = workspace.getFilesAndFolders().map(i => path.basename(i));
+			expect(workspaceBaseNames).toHaveLength(output.size);
+			for (const expectedBaseName of output.keys()) {
+				expect(workspaceBaseNames).toContain(expectedBaseName);
+			}
+		});
+
+		describe('Dotted file/folder handling', () => {
```

Comment:
Since we just got bit by some stuff with dotted files and directories, I added some pretty aggressive test coverage to lock in desired behavior.